tests/e2e/specs/miscellaneous/CustomOpenVSXRegistry.spec.ts

```diff
@@ -0,0 +1,217 @@
+import { DriverHelper } from '../../utils/DriverHelper';
+import { error } from 'selenium-webdriver';
+
+suite(
+	`Create a workspace via launching a factory from the ${FACTORY_TEST_CONSTANTS.TS_SELENIUM_FACTORY_GIT_PROVIDER} repository ${BASE_TEST_CONSTANTS.TEST_ENVIRONMENT}`,
+	function (): void {
+		const browserTabsUtil: BrowserTabsUtil = e2eContainer.get(CLASSES.BrowserTabsUtil);
+		const workspaceHandlingTests: WorkspaceHandlingTests = e2eContainer.get(CLASSES.WorkspaceHandlingTests);
+		const projectAndFileTests: ProjectAndFileTests = e2eContainer.get(CLASSES.ProjectAndFileTests);
+		const loginTests: LoginTests = e2eContainer.get(CLASSES.LoginTests);
+		const pathToPluginRegistry: string = '/projects/devspaces/dependencies/che-plugin-registry/';
+		const internalRegistry: string = 'image-registry.openshift-image-registry.svc:5000';
+		const testWorkspaceUtil: ITestWorkspaceUtil = e2eContainer.get(TYPES.WorkspaceUtil);
+		const driverHelper: DriverHelper = e2eContainer.get(CLASSES.DriverHelper);
+		const viewItems: ViewItem[] | undefined = [];
+		const imageTag: string = 'next';
+		const testRepoProjectName: string = 'devspaces';
+		let currentNamespace: string | undefined = '';
+		let cheClusterNamespace: string | undefined = '';
+		let cheClusterName: string | undefined = '';
+		let projectSection: ViewSection;
+		let kubernetesCommandLineToolsExecutor: KubernetesCommandLineToolsExecutor;
+		let workspaceName: string = '';
+		suiteSetup('Login', async function (): Promise<void> {
+			await loginTests.loginIntoChe();
+		});
+
+		test('Navigate to the factory URL', async function (): Promise<void> {
+			await browserTabsUtil.navigateTo(FACTORY_TEST_CONSTANTS.TS_SELENIUM_FACTORY_URL());
+		});
+
+		test('Obtain workspace name from workspace loader page', async function (): Promise<void> {
+			await workspaceHandlingTests.obtainWorkspaceNameFromStartingPage();
+			workspaceName = WorkspaceHandlingTests.getWorkspaceName();
+			expect(workspaceName, 'Workspace name was not fetched from the loading page').not.undefined;
+		});
+
+		test('Registering the running workspace', function (): void {
+			registerRunningWorkspace(WorkspaceHandlingTests.getWorkspaceName());
+		});
+
+		test('Wait the workspace readiness', async function (): Promise<void> {
+			await projectAndFileTests.waitWorkspaceReadinessForCheCodeEditor();
+		});
+
+		test('Check if a project folder has been created', async function (): Promise<void> {
+			Logger.debug(`new SideBarView().getContent().getSection: get ${testRepoProjectName}`);
+			projectSection = await projectAndFileTests.getProjectViewSession();
+			expect(await projectAndFileTests.getProjectTreeItem(projectSection, testRepoProjectName), 'Project folder was not imported').not
+				.undefined;
+		});
+		test('Accept the project as a trusted one', async function (): Promise<void> {
+			await projectAndFileTests.performTrustAuthorDialog();
+		});
+		test('Edit VSX plugin list', function (): void {
```

Comment:
IMHO, it would make the test more clear if say that the test is removing all VSX extensions from plugin registry, apart from `redhat.vscode-xml` extension.

Reply:
applyed